### PR TITLE
Location registry backend refactoring

### PR DESCRIPTION
**Pull request by lilly-muyama**
- What does this PR do?
This PR created a location-registry microservice and endpoints to be utilized by the location registry.
- What JIRA task is related to this PR?
[1](https://airqoteam.atlassian.net/jira/software/projects/NET/boards/28?selectedIssue=NET-11)
[2](https://airqoteam.atlassian.net/jira/software/projects/NET/boards/28?selectedIssue=NET-23)
[3](https://airqoteam.atlassian.net/jira/software/projects/NET/boards/28?selectedIssue=NET-26)
[4](https://airqoteam.atlassian.net/jira/software/projects/NET/boards/28?selectedIssue=NET-22)
[5](https://airqoteam.atlassian.net/jira/software/projects/NET/boards/28?selectedIssue=NET-12)
[6](https://airqoteam.atlassian.net/jira/software/projects/NET/boards/28?selectedIssue=NET-24)
[7](https://airqoteam.atlassian.net/jira/software/projects/NET/boards/28?selectedIssue=NET-13)
- How do I test out this PR?
Git clone this repo `cd AirQo-api/src/location-registry` Then run `pip install -r requirements.txt` to get all the relevant packages. Afterwards run `flask run` to test out the end points using any REST client tool available to you.
- Which end points should are ready for testing?
partial url: /api/v1/location_registry/create_id
partial url: /api/v1/location_registry/register
partial url: /api/v1/location_registry/locations
partial url: /api/v1/location_registry
partial url: /api/v1/location_registry/edit
partial url: /api/v1/location_registry/update

### CODE REVIEW

**Comment by danielogen** [comment]
> The endpoints are working well. To this test this PR, make sure you have docker installed. Next, navigate to the project root i.e. `cd src/location-registry` then build a docker container: `docker build . -t airqo-location-registry`. This might take some minutes depending on your internet connectivity. Now run app using: `docker run -d -p 5000:8080 airqo-location-registry`. Test the endpoints using any REST client of your choice. The base URL will be: `http://127.0.0.1:5000`.

More on docker commands: https://docs.google.com/document/d/1fqNyprp26BQDgvnStQvGmLfxczaZfiqNNxiQFrWNDTE/edit#

File: src/locate/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Comments in Dockerfiles
+FROM python:3.7
```
Comment (Baalmart) [inline src/locate/Dockerfile:2]: We should not be having file changes inside another microservice which this PR does not address. Noticed changes inside `src/locate` 🤔

File: src/locate/app.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from flask import Flask
+import logging
+import os, sys
+from flask_cors import CORS
+from flask_pymongo import PyMongo
+from controllers.locate import locate_blueprint
```
Comment (Baalmart) [inline src/locate/app.py:6]: same case here....we are seeing changes for the netmanager product yet the PR is for the location registry miscroservice. 🤔

File: src/locate/controllers/locate.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
+from flask import Blueprint, request, jsonify
+from helpers import helper
+import sys
+import ast
```
Comment (Baalmart) [inline src/locate/controllers/locate.py:4]: Same here...changes in the netmanager microservice yet this PR is for `location-registry` 🤔

File: src/locate/controllers/sample.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from flask import Blueprint
+#from helpers import helper
+import sys
```
Comment (Baalmart) [inline src/locate/controllers/sample.py:3]: Same here...changes in netmanager. 🤔

File: src/locate/models/map.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import pandas as pd
+from pymongo import MongoClient
+from dotenv import load_dotenv
+import os #name 'os' is not defined
+
+MONGO_URI = os.getenv('MONGO_URI')
+#_config.MONGO_URI
+
```
Comment (Baalmart) [inline src/locate/models/map.py:8]: same case here....changes in the netmanager microservice. 🤔

File: src/locate/models/parishes.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
+import pandas as pd
+from pymongo import MongoClient
+from dotenv import load_dotenv
+import os
+import sys
+from helpers import helper
+
+MONGO_URI = os.getenv('MONGO_URI')
```
Comment (Baalmart) [inline src/locate/models/parishes.py:8]: Same here.... changes in netmanager 🤔

File: src/locate/models/sample.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+from flask import Blueprint
+#from helpers import helper
+import sys
+
```
Comment (Baalmart) [inline src/locate/models/sample.py:4]: another one..... 🤔

File: src/locate/requirements/common.txt [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+dnspython==1.16.0
+Flask==1.1.2
+Flask-Cors==3.0.8
+Flask-PyMongo==2.3.0
+gunicorn==20.0.4
+itsdangerous==1.1.0
+Jinja2==2.11.2
+joblib==0.14.1
+MarkupSafe==1.1.1
+numpy==1.18.2
```
Comment (Baalmart) [inline src/locate/requirements/common.txt:11]: same case here... 🤔

File: src/locate/requirements/dev.txt [hunk 9]
```diff
@@ -0,0 +1,22 @@
+click==7.1.1
+dnspython==1.16.0
+Flask==1.1.2
+Flask-Cors==3.0.8
+Flask-PyMongo==2.3.0
+gunicorn==20.0.4
```
Comment (Baalmart) [inline src/locate/requirements/dev.txt:6]: another one... 🤔

File: src/locate/requirements/prod.txt [hunk 10]
```diff
@@ -0,0 +1,22 @@
+click==7.1.1
+dnspython==1.16.0
+Flask==1.1.2
+Flask-Cors==3.0.8
+Flask-PyMongo==2.3.0
+gunicorn==20.0.4
+itsdangerous==1.1.0
+Jinja2==2.11.2
+joblib==0.14.1
+MarkupSafe==1.1.1
```
Comment (Baalmart) [inline src/locate/requirements/prod.txt:10]: same case again... 🤔

File: src/location-registry/README.md [hunk 11]
```diff
@@ -1,11 +1,8 @@
 fetch branch from github.
```
Comment (Baalmart) [inline src/location-registry/README.md:2]: I think this one is supposed to be completely deleted and not modified, right?

File: src/locate/helpers/helper.py [hunk 12]
```diff
@@ -8,8 +8,10 @@
 from flask import jsonify
 from shapely.geometry import shape, Point
```
Comment (Baalmart) [inline src/locate/helpers/helper.py:9]: Because of refactoring work, this file might require complete deletion, right?

**Comment by sserurich** [comment]
> The endpoints are working well. To this test this PR, make sure you have docker installed. Next, navigate to the project root i.e. `cd src/location-registry` then build a docker container: `docker build . -t airqo-location-registry`. This might take some minutes depending on your internet connectivity. Now run app using: `docker run -d -p 5000:8080 airqo-location-registry`. Test the endpoints using any REST client of your choice. The base URL will be: `http://127.0.0.1:5000`.

Running endpoints from the docker container hasn't yet succeeded on my local machine(windows laptop). Reviewing issue with @danielogen

**Pull request closed without merging**